### PR TITLE
Builder Refactor

### PR DESCRIPTION
**Pull request by etspaceman**
## Changes Introduced

A short description of the changes that were introduced

## Applicable linked issues

Solves #108 by making the logger configurable via builders in all projects

## Checklist (check all that apply)

- [ ] This change maintains backwards compatibility
- [X] I have introduced tests for all new features and changes
- [x] I have added documentation covering all new features and changes
- [x] This pull-request is ready for review

### CODE REVIEW

File: docs/kpl/ciris.md [hunk 1]
```diff
@@ -60,7 +60,8 @@ All of the configuration below is optional.
 | `KPL_THREADING_MODEL`| `kpl.threading.model` | PER_REQUEST | Sets the threading model that the native process will use. Valid values are PER_REQUEST, POOLED |
 | `KPL_THREAD_POOL_SIZE`| `kpl.thread.pool.size` | | Sets the maximum number of threads that the native process' thread pool will be configured with. Must be a non-negative number. |
 | `KPL_USER_RECORD_TIMEOUT`| `kpl.user.record.timeout` | | Set the value when the user submitted records will be timed out at the Java layer. Valid values are durations (e.g. `1 second`) |
-
+| `KPL_GRACEFUL_SHUTDOWN_FLUSH_ATTEMPTS`| `kpl.graceful.shutdown.flush.attempts` | 5 | Set the amount of times that the KPL will perform a flush() during its graceful shutdown process. |
```
Comment (etspaceman) [inline docs/kpl/ciris.md:63]: Realized we didn't have ciris loaders for the graceful shutdown process. Added those

File: kcl-ciris/src/main/scala/kinesis4cats/kcl/fs2/ciris/KCLCirisFS2.scala [hunk 2]
```diff
@@ -118,29 +124,45 @@ object KCLCirisFS2 {
...
+          F.delay(kinesisClient)
+        )
+      else Resource.pure[F, KinesisAsyncClient](kinesisClient)
+    dClient <-
+      if (managedClients) Resource.fromAutoCloseable(F.delay(dynamoClient))
+      else Resource.pure[F, DynamoDbAsyncClient](dynamoClient)
+    cClient <-
+      if (managedClients)
+        Resource.fromAutoCloseable(F.delay(cloudWatchClient))
+      else Resource.pure[F, CloudWatchAsyncClient](cloudWatchClient)
```
Comment (etspaceman) [inline kcl-ciris/src/main/scala/kinesis4cats/kcl/fs2/ciris/KCLCirisFS2.scala:145]: This and similar approaches were used to create default clients, but also allow users to manage their own resources if needed via setting `managedClients` to false

File: docs/client/circe.md [hunk 3]
```diff
@@ -12,10 +12,9 @@ libraryDependencies += "io.github.etspaceman" %% "kinesis4cats-client-logging-ci
 
 ```scala mdoc:compile-only
 import cats.effect._
-import software.amazon.awssdk.services.kinesis.KinesisAsyncClient
 
 import kinesis4cats.client.KinesisClient
 import kinesis4cats.client.logging.instances.circe._
 
-KinesisClient[IO](KinesisAsyncClient.builder().build(), kinesisClientCirceEncoders)
+KinesisClient.Builder.default[IO].withLogEncoders(kinesisClientCirceEncoders).build
```
Comment (kubukoz) [inline docs/client/circe.md:19]: **suggestion:** maybe `KinesisClientBuilder`?

Comment (etspaceman) [reply docs/client/circe.md:19]: I think I like the `.` here 😄

**Pull request merged**